src/extract_gpuinfo_amdgpu.c

```diff
@@ -301,6 +309,11 @@ static void gpuinfo_amdgpu_shutdown(void) {
   struct gpu_info_amdgpu *allocated, *tmp;
 
   list_for_each_entry_safe(allocated, tmp, &allocations, allocate_list) {
+    if (allocated->hwmonFD >= 0)
+      close(allocated->hwmonFD);
+    if (allocated->sysfsFD >= 0)
+      close(allocated->sysfsFD);
+    fclose(allocated->fanSpeedFILE);
```

Comment:
Hmm, I'm actually relying on the closing of the program to free all the resource usages; it doesn't actually go through the device list to free each device's handles. This allocate_list is more of freeing the array of devices itself.

How I made it work is pre-allocating an array of devices, and when we successfuly get the handles for one device, add that device to the exported linked list. The whole array is added the "allocations" linked list once.

Should we add this free-each-device's handles?

Reply:
I see. Why are you using a list if there is only one element in the "allocations" list, which is static to this file? 

I guess that relying on the program termination to close the file descriptors is fine in that case.

Reply:
Yeah, at first I wasn't sure how many times `get_device_handles` gets called :)